ddtrace/debugging/_function/discovery.py

```diff
@@ -326,11 +326,43 @@ def at_line(self, line: int) -> List[FullyNamedFunction]:
 
         return []
 
+    def _resolve_pair(self, pair: _FunctionCodePair, fullname: str) -> FullyNamedFunction:
+        try:
+            return pair.resolve()
+        except ValueError:
+            # We failed to lookup the function by its code object via the GC.
```

Comment:
It would be valuable to have a few failure test cases to understand what the limitations are here

Reply:
Yeah, you might give it a try with a low threshold, and loop a bunch of times, but if we can't no big deal. Do the exploration tests reproduce it with synapse?

Reply:
> Do the exploration tests reproduce it with synapse?

I don't expect them to because in my tests I was able to resolve the function from the code soon after module import, which is when exploration tests would resolve code objects to functions 🙁